src/core/meshcop/secure_transport.hpp

```diff
@@ -321,6 +321,16 @@ class SecureTransport : public InstanceLocator
      *
      */
     void SetCaCertificateChain(const uint8_t *aX509CaCertificateChain, uint32_t aX509CaCertChainLength);
+
+    /**
+     * Extracts public key from it's own certificate.
+     *
+     * @return public key from own certificate in form of entire ASN.1 field.
+     *
+     */
+
+    mbedtls_asn1_buf *GetOwnPublicKey(void) { return &mOwnCert.pk_raw; }
```

Comment:
Also can declare the method itself as `const`?

Reply:
fixed.